src/components/MainInterface.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import generateToken from '../lib/tokens';
+import { setToken } from '../lib/TokenService';
+import JoinAShoppingList from './JoinAShoppingList';
+
+const MainIterface = ({ setHasToken }) => {
```

Comment:
Oh I see, this is kind of replacing the `home` component, right?  I think it looks so much better now !